student.rb

```diff
@@ -0,0 +1,13 @@
+# Represents a student with specific attributes and behaviors
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(age, classroom, name, parent_permission)
+    super(age, name, parent_permission)
+    @classroom = classroom
+  end
```

Comment:
- [ ] When I try to create an instance of the `Student` class, I get this error:

![image](https://github.com/mosidrum/OOP-school-library/assets/77106962/e6c1bb8e-4165-451f-be0c-d811e4af4523)

---

The `parent_permission` class attribute is passed in as a _**keyword**_ `(parent_permission: true)` argument in the superclass `Person`. Therefore when you call the `super` method in the initialize method of the `Student` class, you need to specify the value that will be passed to the `parent_permission` attribute from the superclass. In this case, you want to pass the value of the `parent_permission` being passed as a parameter when creating an instance of the `Student` class through the `initialize` method to the `parent_permission` attribute when calling super. It is after this that conditions specified in the initialize method of the superclass `Person` will be attainable in the child class `Student`. Kindly take a look at the sample code below 👍 😎:

```ruby
def initialize(age, classroom, name, permission: true)
  super(age, name, parent_permission: permission)
  @classroom = classroom
end
```

Then you can create a new instance of the `Student` class by doing this:

```ruby
student = Student.new(17, 'Music', 'John Doe', parent_permission: false)
```